homeassistant/helpers/device_registry.py

```diff
@@ -168,32 +169,37 @@ async def _async_migrate_func(
         self, old_major_version: int, old_minor_version: int, old_data: dict[str, Any]
     ) -> dict[str, Any]:
         """Migrate to the new version."""
-        if old_major_version < 2 and old_minor_version < 2:
-            # From version 1.1
-            for device in old_data["devices"]:
-                # Introduced in 0.110
-                try:
-                    device["entry_type"] = DeviceEntryType(device.get("entry_type"))
-                except ValueError:
-                    device["entry_type"] = None
-
-                # Introduced in 0.79
-                # renamed in 0.95
-                device["via_device_id"] = device.get("via_device_id") or device.get(
-                    "hub_device_id"
-                )
-                # Introduced in 0.87
-                device["area_id"] = device.get("area_id")
-                device["name_by_user"] = device.get("name_by_user")
-                # Introduced in 0.119
-                device["disabled_by"] = device.get("disabled_by")
-                # Introduced in 2021.11
-                device["configuration_url"] = device.get("configuration_url")
-            # Introduced in 0.111
-            old_data["deleted_devices"] = old_data.get("deleted_devices", [])
-            for device in old_data["deleted_devices"]:
-                # Introduced in 2021.2
-                device["orphaned_timestamp"] = device.get("orphaned_timestamp")
+        if old_major_version < 2:
+            if old_minor_version < 3:
+                # Introduced in 2022.2
+                for device in old_data["devices"]:
+                    device["hw_version"] = device.get("hw_version")
```

Comment:
Maybe move do it chronologically, with the biggest version last?

Reply:
Yes, it doesn't matter for this change, but it may matter for the next one. 👍